PayItForward.Data/Abstraction/IRepository.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace PayItForward.Data
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using PayItForward.Data.Abstraction;
+
+    public interface IRepository<T, TKey> : IDisposable
+        where T : class
```

Comment:
For delete and update, you can have only sync methods as they will only change the entity state

Reply:
Add GetAll method with expression